### ISSUE TITLE
[FIND USAGE] incorrect reference verification for COMPONENT_NAME elements

### ISSUE THREAD

**Issue by vaukalak** [TiVo/intellij-haxe#136]
suppose following code:
```
var foo = new Sprite();
trace(foo);
trace(foo);
```
If you place cursor on second or third line and try to find usages, both usages will be found. But if you'll try to do the same for the first line (in the variable declaration) - no usages will be found.

The problem here is that "var foo" is a reference to PsiElement with token type "COMPONENT_NAME". t
The foo in "trace(foo);", are references to  PsiElement with token type "LOCAL_VAR_DECLARATION_PART", which is parent of element type referenced by "var foo".

In this way elements do not match, and the usages is not found (also refactor->rename doesn't work). I really think that this is a very major bug. 

I've fixed problem in HaxeReferenceImpl::isReferenceTo, by changing the last row 
```
return !chain && resolve == element;
```
to
```
if (element instanceof HaxeComponentName) {
      return !chain && resolve == element.getParent();
    }else {
      return !chain && resolve == element;
    }
```

TIR: STB-9359

but I'm not sure if it's a good way to fix this issue.

**Comment by EBatTiVo** [comment]
Closed inadvertantly.

**Comment by eliasku** [comment]
@vaukalak  It's fixed for next `0.9.10` release

![localvar](https://cloud.githubusercontent.com/assets/3038174/13456342/bfb36aaa-e073-11e5-901e-49c43f29829e.gif)

**Comment by EBatTiVo** [comment]
Since the report is 10 months old and I have no idea if the OP is still watching, I'm going to close it as fixed.  Please re-open if it recurs in 0.9.10.
